### PR TITLE
feat(core): option to disable cache for docker build

### LINKED ISSUE #27916: DockerImageAsset: Support --no-cache
### Describe the feature

Currently there's no option to build docker images without caching. The last comment on https://github.com/aws/aws-cdk/issues/20555 explains that this doesn't function as intended.

### Use Case

I have a `pyproject.toml` that requires installing a package from mainline. It currently has no idea if/when mainline has changed and therefore uses cached layers. I need to manually delete local and ECR layers each time to force rebuild.

### Proposed Solution

Add ability to pass args that don't take values.

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

latest

### Environment details (OS name and version, etc.)

MacOS

### PR DESCRIPTION
**Pull request by msambol**
Closes #27916.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by msambol** [comment]
This touches a few different packages.. I put `feat(core)` but let me know if that's not right.

**Comment by msambol** [comment]
@lpizzinidev Could I get poke you for a review here? 😄

File: packages/aws-cdk-lib/core/lib/assets.ts
Comment (lpizzinidev) [inline packages/aws-cdk-lib/core/lib/assets.ts:1]: We could add a test [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/core/test/assets.test.ts) for this.

File: packages/aws-cdk-lib/cx-api/lib/assets.ts
Comment (lpizzinidev) [inline packages/aws-cdk-lib/cx-api/lib/assets.ts:1]: We could add a test [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-lambda/test/code.test.ts) for this.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Looks good to me 👍
I left some minor suggestions for increasing test coverage.

File: packages/aws-cdk-lib/core/lib/assets.ts
Comment (msambol) [reply packages/aws-cdk-lib/core/lib/assets.ts:1]: added 👍🏼

File: packages/aws-cdk-lib/cx-api/lib/assets.ts
Comment (msambol) [reply packages/aws-cdk-lib/cx-api/lib/assets.ts:1]: added 👍🏼

**Comment by msambol** [comment]
@lpizzinidev Updated with your feedback 👍🏼

**Review by lpizzinidev — APPROVED** [review]

**Review by kaizencc — APPROVED** [review]
Thanks @msambol

**Review by kaizencc — APPROVED** [review]

**Pull request merged**